games/static/games/proto.js

```diff
@@ -274,13 +271,21 @@ class Board {
 
     moveDefender(direction){
         let defender = this.pickDefenderUp()
+        let nextMove = this.calculateNextDefenderMove(direction)
+        this.defenderMove(nextMove)
+        this.putDown(defender)
+    }
+    calculateNextDefenderMove(direction) {
         if (this.defenderCanMoveLeft(direction)) {
-            this.defenderMoveLeft()            
+            return directionLeft            
         }
         if (this.defenderCanMoveRight(direction)) {            
-            this.defenderMoveRight()
-        }        
-        this.putDown(defender)
+            return directionRight
+        }     
+        return 0   
```

Comment:
move this to const with a clear name like NoDirection or something
(it will be wierd to have all the rest as directions and this one without